### PR TITLE
Detailed output for cross-validation. Closes #1708

### LINKED ISSUE #1708: [feature request] add evaluation output for cross-validation
Currently the nlu evaluation script supports different evaluation metrics and outputs (which are very useful :+1:) solely for training- / test-data splits, and not for crossvalidation mode. Those are:
* `--confmat`
* `--errors`
* `--successes`
* `--report`
* `--histogram`

This is partially documented [here](https://rasa.com/docs/nlu/evaluation/#intent-classification), but as an initial step the `evaluate` command should also warn about superfluous arguments, that can not be used, or even abort then.
Even more important would be to support those outputs also for crossvalidation mode. This seems straightforward to me:
* `--confmat`: sum across all cv runs (or average for relative percentages)
* `--errors`: combine across all runs
* `--successes`: combine across all runs
* `--report`: just format the output
* `--histogram`: combine confidences and build histogram on that

It seems to be possible to reuse the functions for normal runs on the combined results of the cv runs.
Did I miss something that makes this more complicated or undesirable? Or is this on your roadmap anyways? Thanks for considering this.

### PR DESCRIPTION
**Pull request by nimf**
**Proposed changes**:
When we run evaluation with a model and test data,
we get many useful things like intents and entities classification reports,
intents confusion matrix, error examples. But when we run evaluation
in cross-validation mode, we only get an overall train/test performance.

In this PR, we collect all test folds predictions
and then use them to produce all the useful information.

This addresses #1708

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for opening this PR, we'll give it a review as soon as possible

File: rasa/nlu/test.py [hunk 1]
```diff
@@ -308,11 +308,11 @@ def evaluate_intents(
             title="Intent Confusion matrix",
             out=confmat_filename,
         )
-        plt.show()
+        plt.show(block=False)
```
Comment (tabergma) [inline rasa/nlu/test.py]: Thanks for changing this! It was also on our list :)

File: rasa/nlu/test.py [hunk 2]
```diff
@@ -352,11 +352,10 @@ def substitute_labels(labels, old, new):
     return [new if label == old else label for label in labels]
 
 
-def evaluate_entities(entity_results, interpreter, report_folder):  # pragma: no cover
+def evaluate_entities(entity_results, extractors, report_folder):  # pragma: no cover
```
Comment (tabergma) [inline rasa/nlu/test.py]: As you touched this file, would you mind adding types? We are cleaning up things on the way. Would be great if you could add types for all the methods you touched.
E.g. update the method definition to
``` 
def evaluate_entities(entity_results: List[EntityEvaluationResult], extractors: Set[Text], report_folder: Optional[Text]):
```

Comment (nimf) [reply rasa/nlu/test.py]: Sure. Will do.

Comment (nimf) [reply rasa/nlu/test.py]: Ok, it's ready for review.

**Pull request merged**